circuits/src/transaction.rs

```diff
@@ -425,18 +315,132 @@ impl<const H: usize, const I: usize> Circuit for TxCircuit<H, I> {
         // Make the payload hash a public input of the circuit
         let payload_hash = composer.append_public(self.payload_hash);
 
-        // Nullify all the tx input notes
-        nullify_gadget::<H, I>(
-            composer,
-            &payload_hash,
-            &self.root,
-            &self.tx_input_notes,
-            &self.tx_output_notes,
-            self.max_fee,
-            self.deposit,
-        )?;
+        // Append the root as public input
```

Comment:
which is the advantage of adding this here and changing the already peer-reviewed comments?

Reply:
Ease of work: the other file was a real pain to debug and I changed only the comments that directly refer to the steps 1 - 6 in the documentation of the function.
I also thought that it is good to have the circuit implementation be more close to the circuit outlined in the specs with clear markers where what things are checked (e.g. membership, ownership, etc)